auslib/web/admin/swagger/api.yaml

```diff
@@ -0,0 +1,441 @@
+  /rules:
+    get:
+      summary: Returns all of Rules in Balrog's DB
+      description: Returns all of the Rules in Balrog’s database inside of a JSON Object.
+      tags:
+        - Rules
+      operationId: auslib.web.admin.views.mapper.rules_get
+      parameters:
+        - name: product
+          in: query
+          description: Name of the product to filter the rules against.
+          type: string
+          x-nullable: true
+          required: false
+        - name: format
+          in: query
+          description: json format applied during testing.
+          type: string
+          x-nullable: true
+          required: false
+      produces: [application/json]
+      deprecated: false
+      responses:
+        '200':
+          description: Get all Rules
+          schema:
+            type: object
+            required:
+              - count
+              - rules
+            properties:
+              count:
+                type: integer
+                format: int32
+                example: 1
+              rules:
+                type: array
+                items:
+                  $ref: '#/definitions/RulesBaseGET'
+                example:
+                  - alias: "esr45-localtest"
+                    backgroundRate: 100
+                    buildID: null
+                    buildTarget: null
+                    channel: "esr-localtest"
+                    comment: "esr45 esr-localtest rule"
+                    data_version: 22
+                    distVersion: null
+                    distribution: null
+                    fallbackMapping: null
+                    headerArchitecture: null
+                    locale: null
+                    mapping: "Firefox-45.7.0esr-build1"
+                    osVersion: null
+                    priority: 43
+                    product: "Firefox"
+                    rule_id: 308
+                    systemCapabilities: null
+                    update_type: "minor"
+                    version: null
+                    whitelist: null
+          #headers : [Optional]
+
+    post:
+      tags:
+        - Rules
+      summary: creates a Rule
+      description: creates a new Rule in Balrog's DB
+      consumes:
+        - application/json
+      produces:
+        - text/html
+      deprecated: false
+      operationId: auslib.web.admin.views.mapper.rules_post
+      parameters:
+        - name: rule
+          in: body
+          description: New Rule object data
+          required: true
+          schema:
+            allOf:
+              - $ref: '#/definitions/CSRFModel'
+              - $ref: '#/definitions/RulesBase'
+
+            # will fail testMissingField test in test_rules.py if test is not modified.
+            # Because only outputs first missing field : backgroundRate
+            # ignores further check i.e. check for priority and returns error.
+            required:
+              - backgroundRate
+              - priority
```

Comment:
We had discussed during the proposal submission phase that we require hierarchical models because , example : Rules model, priority and backgroundRate are both required when creating a new rule, but not required when updating a rule. 
This schema inherits RuleBase and then modifies it to mandate priority and backgroundRate both as required during posting a new rule along with update_type